src/main/java/com/redhat/cloud/notifications/routers/EndpointService.java

```diff
@@ -212,50 +216,41 @@
     }
 
     @PUT
-    @Path("/email/subscription/instant")
+    @Path("/email/subscription/{bundleName}/{applicationName}/{type}")
     @RolesAllowed(RbacIdentityProvider.RBAC_WRITE_INTEGRATIONS_ENDPOINTS)
-    public Uni<Boolean> subscribeInstantEmail(@Context SecurityContext sec) {
+    public Uni<Boolean> subscribeInstantEmail(
+            @Context SecurityContext sec, @PathParam("bundleName") String bundleName, @PathParam("applicationName") String applicationName,
+            @PathParam("type") EmailSubscriptionType type) {
         RhIdPrincipal principal = (RhIdPrincipal) sec.getUserPrincipal();
-        return emailSubscriptionResources.subscribe(
-                principal.getAccount(),
-                principal.getName(),
-                EmailSubscriptionType.INSTANT
-        );
+
+        return applicationResources.getApplication(bundleName, applicationName)
+                .onItem().ifNull().failWith(new NotFoundException())
+                .onItem().transformToUni(application -> emailSubscriptionResources.subscribe(
+                        principal.getAccount(),
+                        principal.getName(),
+                        bundleName,
+                        applicationName,
+                        type
+                ));
     }
 
     @DELETE
-    @Path("/email/subscription/instant")
+    @Path("/email/subscription/{bundleName}/{applicationName}/{type}")
     @RolesAllowed(RbacIdentityProvider.RBAC_WRITE_INTEGRATIONS_ENDPOINTS)
-    public Uni<Boolean> unsubscribeInstantEmail(@Context SecurityContext sec) {
+    public Uni<Boolean> unsubscribeInstantEmail(
```

Comment:
Nit-pick. As the type is passed as parameter, should the method name changed to no longer contain 'instant' ?
Likewise for subscribe above?